flash/core/data/new_data_module.py

```diff
@@ -329,145 +327,3 @@ def on_after_batch_transfer(self, batch: Any, dataloader_idx: int) -> Any:
-
-    @classmethod
-    def create_inputs(
-        cls,
-        enum: Union[LightningEnum, str],
-        train_data: Optional[Any] = None,
-        val_data: Optional[Any] = None,
-        test_data: Optional[Any] = None,
-        predict_data: Optional[Any] = None,
-        train_transform: Optional[INPUT_TRANSFORM_TYPE] = None,
-        val_transform: Optional[INPUT_TRANSFORM_TYPE] = None,
-        test_transform: Optional[INPUT_TRANSFORM_TYPE] = None,
-        predict_transform: Optional[INPUT_TRANSFORM_TYPE] = None,
-        **input_kwarg,
-    ) -> Tuple[Optional[Input]]:
-        cls._verify_input_enum(enum)
-        input_cls: Input = cls.inputs_registry.get(enum)
-        return (
-            cls._create_input(
-                input_cls,
-                train_data,
-                running_stage=RunningStage.TRAINING,
-                transform=train_transform,
-                **input_kwarg,
-            ),
-            cls._create_input(
-                input_cls,
-                val_data,
-                running_stage=RunningStage.VALIDATING,
-                transform=val_transform,
-                **input_kwarg,
-            ),
-            cls._create_input(
-                input_cls,
-                test_data,
-                running_stage=RunningStage.TESTING,
-                transform=test_transform,
-                **input_kwarg,
-            ),
-            cls._create_input(
-                input_cls,
-                predict_data,
-                running_stage=RunningStage.PREDICTING,
-                transform=predict_transform,
-                **input_kwarg,
-            ),
-        )
-
-    @staticmethod
-    def _create_input(
-        input_cls,
-        *load_data_args,
-        running_stage: RunningStage,
-        transform: Optional[InputTransform],
-        **kwargs,
-    ) -> Optional[Input]:
-        if load_data_args[0] is not None:
-            return input_cls(running_stage, *load_data_args, transform=transform, **kwargs)
-
-    @classmethod
-    def _verify_input_enum(cls, enum: LightningEnum) -> None:
-        if not cls.inputs_registry or not isinstance(cls.inputs_registry, FlashRegistry):
-            raise MisconfigurationException(
-                "The ``AutoContainer`` should have ``inputs_registry`` (FlashRegistry) populated "
-                "with Input class and ``default_flash_dataset_enum`` (LightningEnum) class attributes. "
-            )
-
-        if enum not in cls.inputs_registry.available_keys():
-            available_constructors = [f"from_{key.name.lower()}" for key in cls.inputs_registry.available_keys()]
-            raise MisconfigurationException(
-                f"The ``AutoContainer`` ``inputs_registry`` doesn't contain the associated {enum} "
-                f"HINT: Here are the available constructors {available_constructors}"
-            )
-
-    @classmethod
-    def register_input(cls, enum: Union[str, LightningEnum], input_cls: Type[Input]) -> None:
```

Comment:
So we remove entirely the input registry on the DataModule, I think it makes sense IMO.